pkg/cache/objectarraycache/objectarraycache.go

```diff
@@ -0,0 +1,95 @@
+package objectarraycache
+
+import (
+	"context"
+	"time"
+
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+// RefreshFunction is a function that retrieves an array of objects.
+type RefreshFunction[T any] func(ctx context.Context) ([]*T, error)
+
+// ObjectArrayCache is a cache for an array of objects.
+// The object array in the cache instance are valid for a pre-defined period,
+// and are retrieved from the source when they are not valid anymore.
+type ObjectArrayCache[T any] struct {
```

Comment:
I like the concept of this, but I don't necessarily think we should have an object array variant. We could have a generic cache that has expirations on objects with refreshers, and so if you need slice version then you just wrap it from the caller code OR we have a wrapper around the single object that does the wrapping for you

Reply:
Also please add a comment that the elements in the cache are NOT copied, which means that getters of these objects cannot modify them or we may have concurrent read/writes

Reply:
This PR is a quick patch in order to mitigate the customer issue for 4.3. I intend to have a follow-up on this one for 4.4 where a better solution could be used.